src/main/java/com/opentable/kafka/builders/KafkaBaseBuilderFactoryBean.java

```diff
@@ -0,0 +1,43 @@
+package com.opentable.kafka.builders;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import com.codahale.metrics.MetricRegistry;
+
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.opentable.service.AppInfo;
+import com.opentable.service.ServiceInfo;
+import com.opentable.spring.PropertySourceUtil;
```

Comment:
Ok, so I assume you've reviewed the basebuilder, producerbuilder, and consumerBuilder.

let's move on to the next triad

kafkabasebuilderfactorybean
kafkaconsumerfactorybean
kafkaproducerfactorybean